Wrapping/Generators/Python/PyBase/pyBase.i

```diff
@@ -685,6 +685,32 @@ str = str
   }
 %enddef
 
+%define DECL_PYTHON_POINTSETBASE_CLASS(swig_name)
+    %rename(__SetPointsByCoordinates_orig__) swig_name::SetPointsByCoordinates;
+    %extend swig_name {
+        %pythoncode %{
+            def SetPointsByCoordinates(self, points):
+                """Set the points of the pointset by providing their coordinates
+                as a NumPy array."""
+                import numpy as np
+                if hasattr(points, 'ndim') and points.ndim != 1:
+                    self.__SetPointsByCoordinates_orig__(points.ravel())
+                else:
+                    self.__SetPointsByCoordinates_orig__(points)
+
+            def GetPointsByCoordinates(self):
+                """Get the points of the pointset by providing their coordinates
+                as a NumPy array. The array will have a shape of (n_points, dimension)."""
```

Comment:
I'm not sure about the name "GetPointsByCoordinates". Shouldn't it be something like "GetPointsAsNumpyArray"? Or "converts_points_to_np_array"?

Reply:
OK, but we could then also rename the new `SetPointsByCoordinates`. My original intention was to express that `SetPointsByCoordinates` sets the points by specifying a simple sequence of coordinates. For the newly added functionality, `SetPointsByNumpyArray` (or `set_points_by_np_array`) might be a more appropriate name, right?

Reply:
Isn't the API getting unnecessarily complicated?

I thought that you discouraged the use of `SetPoints`?

Reply:
> Isn't the API getting unnecessarily complicated?

When two functions do different things, I think it is clearer to also use different names for those functions. The original `SetPoints(PointsContainer *)` (which has been there for many, many years already, including ITK 1, 2, 3, 4, 5, and 6 alpha 1) _shares_ the container between the caller and the `PointSet`. The new `SetPointsByCoordinates(const std::vector<CoordinateType> &)` _copies_ the coordinates from the caller into the container of the `PointSet`.

> I thought that you discouraged the use of `SetPoints`?

The original `SetPoints(PointsContainer *)` is OK to me. It's OK to "share" your container with a point set, if that's what you want. However, the extra `SetPoints(PointsVectorContainer *)` overload that was added with ITK v5.3.0 appears problematic, as I explained at issue #4848 If I understand correctly, the extra `SetPoints(PointsVectorContainer *)` overload was _specifically_ introduced so support flat coordinate arrays. But unfortunately it may lead to undefined behavior. The new `SetPointsByCoordinates(const std::vector<CoordinateType> &)` was just meant to provide a safe alternative to the problematic `SetPoints(PointsVectorContainer *)` overload of ITK >= v5.3.0.

Is that a helpful explanation to you?

Reply:
@N-Dekker I understand what you mean. However, I do not see the need for another method, which over-complicates the API. It is standard practice in object-oriented APIs to have the same method accept two different types.